frontend/src/hooks/useLogOut.ts

```diff
@@ -0,0 +1,31 @@
+import { useState, useCallback } from 'react';
+import { AuthError } from '@supabase/supabase-js';
+import { useClient } from '@/contexts/AppContext';
+
+interface LogOutResult {
+  isLoading: boolean;
+  error: AuthError | undefined | null;
+}
+
+type LogOutHookFunction = () => Promise<void>;
+
+export type LogOutHook = () => [LogOutHookFunction, LogOutResult];
+
+export const useLogOut: LogOutHook = () => {
+  const client = useClient();
+  const [result, setResult] = useState<LogOutResult>({
+    isLoading: false,
+    error: undefined,
+  });
+
+  const performLogOut = useCallback<LogOutHookFunction>(
+    async () => {
+      setResult({ ...result, isLoading: true });
+      await client.auth.signOut();
```

Comment:
The sign out function does have a chance to return error, so just make changes to take that into account.
```const { error } = await client.auth.signOut();```
```setResult({...result, isLoading: false, error})```

Reply:
Done